### PR TITLE
Visibility lifecycle fixes

### PR DESCRIPTION
**Pull request by kfcampbell**
This PR unifies the work done in [jurgenweber's PR#680](https://github.com/integrations/terraform-provider-github/pull/680) as well as [this test branch](https://github.com/integrations/terraform-provider-github/tree/test/pull%2F680) that some folks were working on.

### CODE REVIEW

**Comment by kfcampbell** [comment]
@jcudit I _think_ this satisfies all of the conditions we talked about earlier this week. What else would you like to see on this PR? I'd be happy to update it with any additional test cases or behavior that you'd like.

File: examples/repo_org_internal/cli.cfg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+provider_installation {
+  filesystem_mirror {
+    path    = "/Users/jurgen.weber/checkouts/terraform/terraform-provider-github/examples/repo_org_internal/terraform.d/plugins"
+    include = ["*/*/*"]
+  }
+  direct {
+    exclude = ["*/*/*"]
+  }
+}
```
Comment (jcudit) [inline examples/repo_org_internal/cli.cfg:9]:
```suggestion
See Building The Provider instructions in CONTRIBUTING.md
```

Can we make this more general by pointing to docs in the repo?

Comment (kfcampbell) [reply examples/repo_org_internal/cli.cfg:9]: Definitely! Would you prefer that we keep this `cli.cfg` file the pointer to the repo docs, or maybe should we add a link to the README in that directory? It feels a little weird to me to keep the `.cfg` extension for a text or markdown file...what do you think?

Comment (jcudit) [reply examples/repo_org_internal/cli.cfg:9]: > add a link to the README in that directory

Good catch!  My vote goes to adding a link.  https://github.com/integrations/terraform-provider-github/blob/master/CONTRIBUTING.md#building-the-provider addresses this indirectly, so maybe we could link to there.

Comment (kfcampbell) [reply examples/repo_org_internal/cli.cfg]: Done! I've removed the config and provided the link [here](https://github.com/integrations/terraform-provider-github/pull/746/commits/b94f595dca2dd56a6601e275d7feef31fbbd9e2e), and I've also renamed the example directory and created repo name to match the existing pattern [in this commit](https://github.com/integrations/terraform-provider-github/pull/746/commits/303db1f3e317e0bb9b17fd28029d13b3a1c93d00). 

How does that look?

Comment (jcudit) [reply examples/repo_org_internal/cli.cfg]: Looking great 🚀

**Review by jcudit — APPROVED** [review]

**Pull request merged**